src/operator/reduce.ts

```diff
@@ -53,16 +53,24 @@ export function reduce<T>(this: Observable<T>, accumulator: (acc: T[], value: T,
 export function reduce<T, R>(this: Observable<T>, accumulator: (acc: R, value: T, index: number) => R, seed?: R): Observable<R>;
 /* tslint:disable:max-line-length */
 export function reduce<T, R>(this: Observable<T>, accumulator: (acc: R, value: T) => R, seed?: R): Observable<R> {
-  return this.lift(new ReduceOperator(accumulator, seed));
+  let hasSeed = false;
+  // providing a seed of `undefined` *should* be valid and trigger
+  // hasSeed! so don't use `seed !== undefined` checks!
+  // For this reason, we have to check it here at the original call site
+  // otherwise inside Operator/Subscriber we won't know if `undefined`
+  // means they didn't provide anything or if they literally provided `undefined`
+  if (arguments.length >= 2) {
+    hasSeed = true;
+  }
+
+  return this.lift(new ReduceOperator(accumulator, seed, hasSeed));
 }
 
 export class ReduceOperator<T, R> implements Operator<T, R> {
-
-  constructor(private accumulator: (acc: R, value: T) => R, private seed?: R) {
-  }
+  constructor(private accumulator: (acc: R, value: T) => R, private seed?: R, private hasSeed: boolean = false) {}
 
   call(subscriber: Subscriber<R>, source: any): any {
-    return source._subscribe(new ReduceSubscriber(subscriber, this.accumulator, this.seed));
+    return source._subscribe(new ReduceSubscriber(subscriber, this.accumulator, this.seed, this.hasSeed));
```

Comment:
@trxcllnt sorry, I'm not following what your suggestion is? Is it to move the `arguments.length >= 2` check into the `ScanOperator` constructor and use `new ScanOperator(...arguments)`?

A little pedantic of me, but technically that will perform not as well because `new ScanOperator(...arguments)` transpiles to something like `new (Function.prototype.bind.apply(Wow, [null].concat(args)))();`. But this really isn't the typical hot code path, so prolly doesn't matter.